src/main/java/com/wcc/platform/service/ProgrammeService.java

```diff
@@ -0,0 +1,41 @@
+package com.wcc.platform.service;
+
+import static com.wcc.platform.domain.cms.ApiResourcesFile.PROG_BOOK_CLUB;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.wcc.platform.domain.cms.attributes.ProgramType;
+import com.wcc.platform.domain.cms.pages.programme.ProgrammePage;
+import com.wcc.platform.domain.exceptions.PlatformInternalException;
+import com.wcc.platform.utils.FileUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+/** Programme Service. */
+@Service
+public class ProgrammeService {
+
+  private final ObjectMapper objectMapper;
+
+  @Autowired
+  public ProgrammeService(final ObjectMapper objectMapper) {
+    this.objectMapper = objectMapper;
+  }
+
+  /**
+   * API to fetch details about different type of programmes.
+   *
+   * @return Programme Page json response
+   */
+  public ProgrammePage getProgramme(final ProgramType programType) {
+    try {
+      String data = null;
+      if (ProgramType.BOOK_CLUB.equals(programType)) {
+        data = FileUtil.readFileAsString(PROG_BOOK_CLUB.getFileName());
+      }
+      return objectMapper.readValue(data, ProgrammePage.class);
+    } catch (JsonProcessingException e) {
+      throw new PlatformInternalException(e.getMessage(), e);
```

Comment:
also please remember to add the missing test for the events. I tried to add some, but I was not able to finish all.